projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1628,8 +1167,31 @@
 			} );
 
 			carousel.gallery.innerHTML = '';
+
 			initCarouselSlides( gallery.querySelectorAll( settings.imgSelector ), settings.startIndex );
-			selectSlideAtIndex( settings.startIndex );
+			carousel.overlay.style.display = 'block';
+
+			swiper = new Swiper( '.swiper-container', {
+				centeredSlides: true,
+				pagination: {
+					el: '.swiper-pagination',
+				},
+				navigation: {
+					nextEl: '.swiper-button-next',
+					prevEl: '.swiper-button-prev',
+				},
+				initialSlide: settings.startIndex,
+				on: {
+					init: function () {
+						selectSlideAtIndex( settings.startIndex );
+					},
+				},
+			} );
+
+			swiper.on( 'slideChange', function () {
+				selectSlideAtIndex( swiper.activeIndex );
+			} );
+			carousel.overlay.style.opacity = '1';
```

Comment:
should be able to use existing fadeIn method for this but wasn't working for some reason, so temporarily adjusting here and will come back to this.

Reply:
Let me know if you need a hand with that when you look into it later. I wrote that method to replace `jQuery` functionality in a more performant way, but I could have missed something.

Reply:
This was my mistake - had set a `display:none` in css in stead as inline style - fixed now.